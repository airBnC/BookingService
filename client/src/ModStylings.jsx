import styled from 'styled-components';
import logo from './images/arrowLogo.png';
import arrow from './images/arrow.png';

export const Holder = styled.div`
display: flex;
flex-direction: column;
width: inherit;
padding-top: 10px;
`;

export const InputHolder = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border: 1px solid rgb(172, 172, 172);
  padding: 5px;
  padding-left: 15px;
  padding-right: 10px;
  &:hover{
    cursor: pointer;
  }
`;

export const Title = styled.div`
  font-size: 12px;
  font-weight: bold;
  margin-bottom: 3px;
`;

export const Logo = styled.div`
  content: url(${logo});
  height: 17px;
  width: 25px;
  transform: rotate(${props => (props.clicked ? '0turn' : '0.5turn')});
`;

export const Arrow = styled.div`
  content: url(${arrow});
  height: 17px;
`;

export const DropdownHolder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  user-select: none;
`;

export const Close = styled.button`
  margin-top: 8px;
  border: none;
  align-self: flex-end;
  font-size: 14px;
  color: #007D8C;
  outline: none;
  &:hover{
    text-decoration: underline;
  }
  &:active{
    color: #FD5C63;
  }
`;
