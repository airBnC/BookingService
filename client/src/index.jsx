import React, { Component } from 'react';
import { render } from 'react-dom';
import styled, { injectGlobal } from 'styled-components';
import axios from 'axios';
import ModOne from './pricing_mod1/ModOne';
import ModTwo from './calendar_mod2/ModTwo';
import ModThree from './guests_mod3/ModThree';
import { setStartOrEndDate, clearDates, calendarChange } from './calendar_mod2/CalendarLogic';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: 300px;
  border: 1px solid rgb(172, 172, 172);
  font-family: Quicksand;
  padding: 24px;
`;

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      adults: 1,
      children: 0,
      infants: 0,
      totalGuests: 1,
      listingData: null,
      startDate: null,
      endDate: null,
      mod2Clicked: 'closed',
      mod3Clicked: false,
    };
    this.setStartOrEndDate = setStartOrEndDate.bind(this);
    this.clearDates = clearDates.bind(this);
    this.calendarChange = calendarChange.bind(this);
  }

  componentDidMount() {
    this.getListingData();
  }

  onGuestButtonClick = (guest, increment) => {
    const val = this.state[guest] + increment;
    const { totalGuests } = this.state;
    const total = totalGuests + increment;
    const { maxGuests } = this.state.listingData;
    // infants don't count towards the number of guests, but have a cap of 5
    if (guest === 'infants' && (val > 5 || val < 0)) {
      return;
    // there is a minimum of 1 adult guest
    } else if (guest === 'adults' && (val > maxGuests || val < 1 || total > maxGuests)) {
      return;
    } else if (guest === 'children' && (val > maxGuests || val < 0 || total > maxGuests)) {
      return;
    }
    this.setState({
      [guest]: val,
      totalGuests: guest === 'infants' ? totalGuests : total,
    });
  }

  onMod3InputHolderClick = () => {
    this.setState({
      mod3Clicked: !this.state.mod3Clicked,
    });
  }

  getListingData = () => {
    let id = window.location.pathname;
    if (id === '/') {
      id = 1;
    } else {
      id = id.replace(/\//g, '');
    }
    axios.get(`http://127.0.0.1:3002/rooms/${id}/bookingInfo/`)
      .then((response) => {
        this.setState({
          listingData: response.data,
        });
      })
      .catch(() => {
        console.log('there was an error!');
      });
  }

  render() {
    if (this.state.listingData) {
      return (
        <Holder>
          <ModOne
            price={this.state.listingData.pricePerNight}
            rating={this.state.listingData.starRating}
            numReviews={this.state.listingData.custRevNum}
          />
          <ModTwo
            dates={this.state.listingData.datesTaken}
            minStay={this.state.listingData.minStay}
            startDate={this.state.startDate}
            endDate={this.state.endDate}
            setDate={this.setStartOrEndDate}
            calendarChange={this.calendarChange}
            clicked={this.state.mod2Clicked}
            clearDates={this.clearDates}
          />
          <ModThree
            adult={this.state.adults}
            child={this.state.children}
            infant={this.state.infants}
            totalGuests={this.state.totalGuests}
            btnClick={this.onGuestButtonClick}
            maxGuests={this.state.listingData.maxGuests}
            clicked={this.state.mod3Clicked}
            close={this.onMod3InputHolderClick}
          />
          {/* <ModFour /> */}
        </Holder>
      );
    }
    return null;
  }
}

render(<App />, document.getElementById('root'));

// cleaningFee={this.state.listingData.cleaningFee}
// maxGuests={this.state.listingData.maxGuests}
// serviceFee={this.state.listingData.serviceFee}
