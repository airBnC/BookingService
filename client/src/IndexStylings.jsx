import styled, { injectGlobal } from 'styled-components';
import { abnbGrey, abnbRed } from './ModStylings';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  user-select: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: ${props => ((props.startDate && props.endDate) ? '400px' : '300px')};
  border: 1px solid ${abnbGrey};
  font-family: Roboto, sans-serif;
  padding: 24px;
`;

export const Button = styled.button`
  width: inherit;
  margin-top: 20px;
  height: 40px;
  border-radius: 7px;
  outline: none;
  background-color: ${abnbRed};
  font-weight: bold;
  color: white;
`;

export const Details = styled.div`
  font-size: 12px;
  font-weight: bold;
  align-self: center;
  margin-top: 7px;
`;
