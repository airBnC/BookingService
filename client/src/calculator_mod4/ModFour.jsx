import React from 'react';
import PropTypes from 'prop-types';
import { HolderMod4 } from '../ModStylings';

export default function ModFour(props) {
  if (props.startDate && props.endDate) {
    const numNights = props.endDate - props.startDate;
    const priceOfStay = numNights * props.price;
    const total = priceOfStay + props.cleaningFee + props.serviceFee;
    return (
      <HolderMod4 main>
        <HolderMod4>
          <div> ${props.price} X {numNights} </div>
          <div> ${priceOfStay} </div>
        </HolderMod4>
        <HolderMod4>
          <div> Cleaning Fee </div>
          <div> ${props.cleaningFee}</div>
        </HolderMod4>
        <HolderMod4>
          <div> Service Fee </div>
          <div> ${props.serviceFee} </div>
        </HolderMod4>
        <HolderMod4 total>
          <div> Total </div>
          <div> ${total} </div>
        </HolderMod4>
      </HolderMod4>
    );
  }
  return null;
}

ModFour.propTypes = {
  startDate: PropTypes.number,
  endDate: PropTypes.number,
  price: PropTypes.number,
  cleaningFee: PropTypes.number,
  // maxGuests: PropTypes.number,
  serviceFee: PropTypes.number,
};

ModFour.defaultProps = {
  startDate: 0,
  endDate: 0,
  price: 0,
  cleaningFee: 0,
  // maxGuests: 0,
  serviceFee: 0,
};
