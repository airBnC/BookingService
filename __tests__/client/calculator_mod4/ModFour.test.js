import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ModFour from '../../../client/src/calculator_mod4/ModFour';
import { HolderMod4 } from '../../../client/src/ModStylings';

configure({ adapter: new Adapter() });

const props = {
  startDate: 1,
  endDate: 4,
  price: 51,
  cleaningFee: 20,
  serviceFee: 20,
};

describe('mod4 component holder', () => {
  it('is the main flex component', () => {
    const tree = renderer
      .create(<HolderMod4 mod1> things inside </HolderMod4>)
      .toJSON();
    expect(tree).toHaveStyleRule('display', 'flex');
  });
});

describe('mod4 component', () => {
  it('renders the calculator', () => {
    const tree = shallow(<ModFour {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('does not render if rating is undefined', () => {
    const tree = renderer
      .create(<ModFour checkIn={null} checkOut={null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
