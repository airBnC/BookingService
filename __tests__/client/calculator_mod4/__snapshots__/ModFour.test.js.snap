// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mod4 component does not render if rating is undefined 1`] = `null`;

exports[`mod4 component renders the calculator 1`] = `
<ModStylings__HolderMod4
  main={true}
>
  <ModStylings__HolderMod4>
    <div>
       $
      51
       X 
      3
       
    </div>
    <div>
       $
      153
       
    </div>
  </ModStylings__HolderMod4>
  <ModStylings__HolderMod4>
    <div>
       Cleaning Fee 
    </div>
    <div>
       $
      20
    </div>
  </ModStylings__HolderMod4>
  <ModStylings__HolderMod4>
    <div>
       Service Fee 
    </div>
    <div>
       $
      20
       
    </div>
  </ModStylings__HolderMod4>
  <ModStylings__HolderMod4
    total={true}
  >
    <div>
       Total 
    </div>
    <div>
       $
      193
       
    </div>
  </ModStylings__HolderMod4>
</ModStylings__HolderMod4>
`;
